node/src/components/chain_synchronizer/operations.rs

```diff
@@ -532,102 +493,209 @@ async fn sync_trie_store(
+    effect_builder: EffectBuilder<JoinerEvent>,
+    chainspec: &Chainspec,
+    trusted_block_header: &BlockHeader,
+    trusted_key_block_info: &KeyBlockInfo,
+) -> Result<BlockHeader, Error> {
+    let mut most_recent_block_header = trusted_block_header.clone();
+    let mut current_trusted_key_block_info = trusted_key_block_info.clone();
     info!("start - fetch block headers up to the most recent one - fast sync");
     loop {
         let maybe_fetched_block = fetch_and_store_next::<BlockHeaderWithMetadata>(
             effect_builder,
             &most_recent_block_header,
-            &trusted_key_block_info,
+            &current_trusted_key_block_info,
             chainspec,
         )
         .await?;
-        match maybe_fetched_block {
-            Some(more_recent_block_header_with_metadata) => {
-                most_recent_block_header = more_recent_block_header_with_metadata.block_header;
-                // If the new block is a switch block, update the validator weights, etc...
-                if let Some(key_block_info) = KeyBlockInfo::maybe_from_block_header(
-                    &most_recent_block_header,
-                    chainspec.protocol_config.verifiable_chunked_hash_activation,
-                ) {
-                    trusted_key_block_info = key_block_info;
-                }
+
+        if let Some(more_recent_block_header_with_metadata) = maybe_fetched_block {
+            most_recent_block_header = more_recent_block_header_with_metadata.block_header;
+
+            // TODO: If the new block is a switch block, update the validator weights, etc...
+            if let Some(key_block_info) = KeyBlockInfo::maybe_from_block_header(
+                &most_recent_block_header,
+                chainspec.protocol_config.verifiable_chunked_hash_activation,
+            ) {
+                current_trusted_key_block_info = key_block_info;
             }
+        } else {
             // If we timed out, consider syncing done.
-            None => break,
+            break;
         }
+
         // If we synced up to the current era, we can also consider syncing done.
         if is_current_era(
             &most_recent_block_header,
-            &trusted_key_block_info,
+            &current_trusted_key_block_info,
             chainspec,
         ) {
             break;
         }
     }
     debug!("finish - fetch block headers up to the most recent one - fast sync");
+    Ok(most_recent_block_header)
+}
 
-    // Fetch and store all blocks that can contain not-yet-expired deploys. These are needed for
-    // replay detection.
+/// Fetch and store all blocks that can contain not-yet-expired deploys. These are needed for
+/// replay detection.
+async fn fetch_blocks_for_deploy_replay_protection(
+    effect_builder: EffectBuilder<JoinerEvent>,
+    chainspec: &Chainspec,
+    most_recent_block_header: &BlockHeader,
+    trusted_key_block_info: &KeyBlockInfo,
```

Comment:
…so we could probably safely use `most_recent_key_block_info` here instead.
(I guess it's not a blocker: We'll just fetch more blocks than necessary this way.)

Reply:
Resolved in c33e3e886d0af356d770003062fed1a00114292d